frontend/public/components/service-catalog/create-instance.tsx

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-undef */
+
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Helmet } from 'react-helmet';
+import { Link } from 'react-router-dom';
+
+import { LoadingBox } from '../utils/status-box';
+import { history, Firehose, kindObj, NavTitle, resourcePathFromModel } from '../utils';
+import { ClusterServiceClassModel, ServiceInstanceModel } from '../../models';
+import { k8sCreate, serviceClassDisplayName } from '../../module/k8s';
+import { NsDropdown } from '../RBAC/bindings';
+import { ClusterServiceClassInfo } from '../cluster-service-class-info';
+import { ButtonBar } from '../utils/button-bar';
+
+const getAvailablePlans = (plans: any[]): any[] => _.reject(_.get(plans, 'data'), plan => plan.status.removedFromBrokerCatalog);
+
+class CreateInstanceForm extends React.Component<CreateInstanceFormProps, CreateInstanceFormState> {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      name: '',
+      namespace: '',
+      plan: '',
+      inProgress: false,
+    };
+
+    this.onNamespaceChange = this.onNamespaceChange.bind(this);
+    this.onNameChange = this.onNameChange.bind(this);
+    this.save = this.save.bind(this);
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    const { name, plan } = state;
+    const { obj, plans } = props;
+
+    const defaultName = _.get(obj, 'data.spec.externalName');
+    if (!name && defaultName) {
+      return { name: defaultName };
+    }
+
+    const defaultPlan = _.get(getAvailablePlans(plans), '[0].spec.externalName');
```

Comment:
So if no `!name && defaultName` return {...}, does this mean `plan` will not be executed/returned?